pkg/model/estimate.go

```diff
@@ -0,0 +1,99 @@
+package model
+
+import (
+    "os/exec"
+	"net"
+	"encoding/json"
+	"log"
+	// "os"
+)
+
+const (
+	SERVE_SOCKET = "/tmp/estimator.sock"
+)
+
+type Estimator struct {
+	Err chan error	
+	*exec.Cmd
+}
+
+type PowerRequest struct {
+	ModelName string   `json:"model_name"`
+	XCols 	[]string 	`json:"x_cols"`
+	XValues [][]float32 `json:"x_values"`
+	CorePower []float32 `json:"core_power"`
+	DRAMPower []float32 `json:"dram_power"`
+	GPUPower []float32 `json:"gpu_power"`
+	OtherPower []float32 `json:"other_power"`
+}
+
+type PowerResponse struct {
+	Powers []float32   `json:"powers"`
+	Message string  	`json:"msg"`
+}
+
+func (e *Estimator) StartPyEstimator() {
+    e.Cmd = exec.Command("python3", "./py/estimator.py")
+	go func() {
+		e.Err <- e.Cmd.Start()
+	}()
+	select{
+		case err := <- e.Err:
+			log.Printf("error: %v", err)
+	}
+}
+
+func (e *Estimator) Destroy() {
+	if err := e.Cmd.Process.Kill(); err != nil {
+		log.Printf("failed to kill process: %v", err)
+	} else {
+		log.Printf("kill estimator")
+	}
+}
+
+
+func (e *Estimator) GetPower(modelName string, xCols []string, xValues [][]float32, corePower, dramPower, gpuPower, otherPower []float32) []float32 {
```

Comment:
Did not find where this function is called
Will it replace some code in `pkg/collector/reader.go`?

Reply:
No. I haven't replaced it the reader.go yet. 
Need to convert the current PodEnergy to xCols, xValues and Package power to corePower, dramPower, and so on.

Reply:
Ahhh ok, this PR is draft then!